### PR TITLE
OPNET-197: Remove node-ip from kubelet for dual-stack vSphere

### PR DESCRIPTION
**Pull request by mkowalski**
This PR modifies the kubelet systemd unit for vSphere platform so that in case of dual-stack networking we do not pass `--node-ip` parameter.

This change is part of onboarding vSphere platform to dual-stack networking but due to the fact that it's currently impossible to pass dual-stack IPs as `--node-ip` and to use `--cloud-provider` at the same time, we are using a temporary workaround that removes explicit node IP setting and lets kubelet pick it automatically.

This change will no longer be needed when the upstream enhancement https://github.com/kubernetes/enhancements/pull/3706 gets implemented, but currently seems like the most reasonable solution to enable such a setup.

This change creates a limitation that vSphere dual-stack cluster should be used only on a single NIC configurations due to the fact that we will be relying on the kubelet logic to select IP addresses and this proven problematic in the past. In order to avoid kubelet selecting wrong interfaces, we want customers to only use this configuration with a single NIC in order to reduce the scope of kubelet's possible mistakes.

Closes: [OPNET-197](https://issues.redhat.com//browse/OPNET-197)

/cc @cybertron

### CODE REVIEW

**Comment by mkowalski** [comment]
The newly created files are copies of `templates/master/01-master-kubelet/_base/units/kubelet.service.yaml` and `templates/worker/01-worker-kubelet/_base/units/kubelet.service.yaml` respectively with the only real change being

```
{{- if ne .IPFamilies "DualStack"}}
        --node-ip=${KUBELET_NODE_IP} \
{{- end}}
```

so that for single-stack we pass `--node-ip` as usual but for dual-stack we remove it completely.

**Comment by dougsland** [comment]
ci not seems related to the patch: `{  failed to acquire lease for "alibabacloud-quota-slice": status 502 Bad Gateway, status code 502}`

**Comment by mkowalski** [comment]
/test e2e-vsphere

Because of the location of the change, I am not really worried about non-vsphere jobs failing here

**Comment by mkowalski** [comment]
/hold

For a moment, because I'm investigating deeper the k/cloud-provider code in respect to https://github.com/kubernetes/cloud-provider/blob/bf769589bdb472d92d86b29eb4016a312d4b5ca8/node/helpers/address.go#L47-L57

TLDR: passing for dual-stack always `--node-ip 0.0.0.0` instead of `--node-ip 1.1.1.1` may be another solution here; not necessarily better, but I want to check it out before we make a final decision

**Comment by mkowalski** [comment]
/cc rphillips sjenning mrunalp

For approval as kubelet template owners

**Review by dougsland — COMMENTED** [review]
/lgtm

**Review by sinnykumari — APPROVED** [review]
LGTM
vsphere test is green

**Pull request merged**
